services/server/config/config.go

```diff
@@ -435,6 +432,46 @@ func mergeConfig(to, from *Config) error {
 	return nil
 }
 
+type sliceTransformer struct{}
+
+func (sliceTransformer) Transformer(t reflect.Type) func(dst, src reflect.Value) error {
+	if t.Kind() != reflect.Slice {
+		return nil
+	}
+	return func(dst, src reflect.Value) error {
+		if !dst.CanSet() {
+			return nil
+		}
+		if src.Type() != dst.Type() {
+			return fmt.Errorf("cannot append two slice with different type (%s, %s)", src.Type(), dst.Type())
+		}
+		for i := 0; i < src.Len(); i++ {
+			found := false
+			for j := 0; j < dst.Len(); j++ {
+				srcv := src.Index(i)
+				dstv := dst.Index(j)
+				if !srcv.CanInterface() || !dstv.CanInterface() {
+					if srcv.Equal(dstv) {
+						found = true
+						break
+					}
+				} else if reflect.DeepEqual(srcv.Interface(), dstv.Interface()) {
+					found = true
+					break
+				} else {
+					fmt.Printf("not equal: %#v vs %#v\n", srcv.Interface(), dstv.Interface())
+				}
+			}
+			if !found {
+				fmt.Printf("not found, appending %#v\n", src.Index(i).Interface())
```

Comment:
Can we pass down a logger from the context of the caller into `sliceTransformer`?

Reply:
Left over debug lines, I just removed them instead.